source/postprocess/depth_average.cc

```diff
@@ -73,25 +73,87 @@ namespace aspect
+          for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+            variables.push_back(std::string("C_") + Utilities::int_to_string(c));
+          variables.push_back ("adiabatic_temperature");
+          variables.push_back ("velocity_magnitude");
+          variables.push_back ("sinking_velocity");
+          variables.push_back ("Vs");
+          variables.push_back ("Vp");
+          variables.push_back ("viscosity");
+          variables.push_back ("vertical_heat_flux");
+          n_statistics = variables.size();
+        }
+      else
+        {
+          for ( std::vector<std::string>::const_iterator it = output_variables.begin();
+                it != output_variables.end(); ++it)
+            {
+              if ( *it == "composition")
+                {
+                  n_statistics += this->n_compositional_fields();
+                  for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+                    variables.push_back(std::string("C_") + Utilities::int_to_string(c));
+                }
+              else
+                {
+                  n_statistics++;
+                  variables.push_back(*it);
+                }
+            }
+        }
+      Assert (variables.size() == n_statistics, ExcInternalError());
 
       DataPoint data_point;
       data_point.time       = this->get_time();
       data_point.values.resize(n_statistics, std::vector<double> (n_depth_zones));
 
-      // add temperature and the compositional fields that follow
-      // it immediately
+      //Add all the requested fields
       {
-        this->get_lateral_averaging().get_temperature_averages(data_point.values[0]);
-        for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
-          this->get_lateral_averaging().get_composition_averages(c, data_point.values[1+c]);
-        this->get_adiabatic_conditions().get_adiabatic_temperature_profile(data_point.values[1+this->n_compositional_fields()]);
-        this->get_lateral_averaging().get_velocity_magnitude_averages(data_point.values[2+this->n_compositional_fields()]);
-        this->get_lateral_averaging().get_sinking_velocity_averages(data_point.values[3+this->n_compositional_fields()]);
-        this->get_lateral_averaging().get_Vs_averages(data_point.values[4+this->n_compositional_fields()]);
-        this->get_lateral_averaging().get_Vp_averages(data_point.values[5+this->n_compositional_fields()]);
-        this->get_lateral_averaging().get_viscosity_averages(data_point.values[6+this->n_compositional_fields()]);
-        this->get_lateral_averaging().get_vertical_heat_flux_averages(data_point.values[7+this->n_compositional_fields()]);
+        unsigned int index = 0;
+
+        //temperature
+        if ( output_all_variables || std::find( output_variables.begin(), output_variables.end(), "temperature") != output_variables.end() )
+          this->get_lateral_averaging().get_temperature_averages(data_point.values[index++]);
```

Comment:
example: change your test to output temperature last ("velocity_magnitude, temperature") and you will see that the column name is wrong.